### PR TITLE
[flutter_tools] do not try to build tool from dart.sh

### LINKED ISSUE #121894: [flutter_tools] if flutter_tools code doesn't compile, cannot invoke //flutter/bin/dart
Even when just invoking the dart wrapper bash script in //flutter/bin/dart, we run the same logic to verify the flutter CLI tool is up to date and pre-compiled as a JIT snapshot.

We actually don't need to do this if the user is only wanting to invoke the Dart CLI tool. And, for this particular user, it means they cannot use the dart analyzer/language server if they are in the middle of refactoring the tool and the code does not compile.

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/121894

### CODE REVIEW

**Comment by christopherfujino** [comment]
TODO add a test

**Comment by christopherfujino** [comment]
friendly ping, please write a test

**Comment by eliasyishak** [comment]
Friendly ping from PR review @christopherfujino

**Review by andrewkolos — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/test/integration.shard/bash_entrypoint_test.dart [hunk 1]
```diff
@@ -48,6 +48,64 @@ Future<void> main() async {
...
+      // copy dart and shared.sh to temp dir
+      final File trueSharedSh = flutterRoot.childDirectory('bin').childDirectory('internal').childFile('shared.sh');
+      final File fakeSharedSh = (tempDir.childDirectory('bin').childDirectory('internal')
+          ..createSync(recursive: true))
+          .childFile('shared.sh');
+      trueSharedSh.copySync(fakeSharedSh.path);
+      final File fakeDartBash = tempDir.childDirectory('bin').childFile('dart');
+      dartBash.copySync(fakeDartBash.path);
+      // mark dart executable
+      makeExecutable(fakeDartBash);
```
[the remark this answers is not in the document]
Comment (andrewkolos) [reply packages/flutter_tools/test/integration.shard/bash_entrypoint_test.dart:70]: Feel free to disregard. This is a tiny thing to delay merging for

[the remark this answers is not in the document]
Comment (christopherfujino) [reply packages/flutter_tools/test/integration.shard/bash_entrypoint_test.dart:70]: done

**Pull request merged**